pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +384,15 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pods []*k8sv1.P
 		return nil
 	} else if isPodReady(pod) && !isPodOwnedByHandler(pod) {
 		pod := pod.DeepCopy()
+
+		// fail vmi creation if CPU pinning has been requested but the Pod QOS is not Guaranteed
+		podQosClass := qos.GetPodQOS(pod)
```

Comment:
Can we check it on the validation stage?

Reply:
I think it's important to the check this when the pod is running.
After the validation there are changes in the template, then there can limitrange added to the namespace and all kind of admission controllers.
I think we shouldn't start the vm if we didn't get the right QOS.

Reply:
will move it to validation after adding mutating webhook :smile:

Reply:
@vladikr I think that just checking the `status.qosClass` field on the pod should be good enough.